### PR TITLE
Optimize GridFS throughput by removing redundant byte array cloning.

### PR DESCRIPTION
**Pull request by vbabanin**
### Description
This PR introduces performance improvement in the GridFS implementation of the sync Java driver, specifically targeting the `GridFSDownloadStream` and `GridFSUploadStream` classes. It was identified that byte arrays are cloned twice internally via the `Binary` class, which impacts throughput.

## Key Changes:
Modified `GridFSDownloadStream` and `GridFSUploadStream` to use `BsonDocument `instead of `Document.` The `BsonDocument` implementation avoids the unnecessary copying in the `BsonBinary` class, thus enhancing performance.

## Performance Improvements:
**GridFS Download:**
Stable Region Mean on main branch = 769.21 MBps
Perf result mean = 1030.63 MBps
Improvement: Approximately 34.02% increase in throughput

**GridFS Upload:**
Stable Region Mean on main branch = 450.62 MBps
Perf result mean = 515.84 MBps
Improvement: Approximately 14.48% increase in throughput

## Additional Information:
For further details and results of performance tests executed on this patch, please refer to the [performance test results](https://spruce.mongodb.com/task/mongo_java_driver_perf_perf_patch_339bd2cabe4baa7a624588978400a0777d257e9b_665f022c98f4ac0007ae6873_24_06_04_12_01_50/trend-charts?execution=0).

JAVA-5485

### CODE REVIEW

File: driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java [hunk 1]
```diff
@@ -232,28 +233,28 @@ private Document getChunk(final int startChunkIndex) {
...
         }
-        return findIterable.batchSize(batchSize).sort(new Document("n", 1)).iterator();
+        return findIterable.batchSize(batchSize).sort(new BsonDocument("n", new BsonInt32(1))).iterator();
     }
 
-    private byte[] getBufferFromChunk(@Nullable final Document chunk, final int expectedChunkIndex) {
+    private byte[] getBufferFromChunk(@Nullable final BsonDocument chunk, final int expectedChunkIndex) {
 
-        if (chunk == null || chunk.getInteger("n") != expectedChunkIndex) {
+        if (chunk == null || chunk.get("n").asInt32().getValue() != expectedChunkIndex) {
```
Comment (jyemin) [inline driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java:249]:
```suggestion
        if (chunk == null || chunk.getInt32("n").getValue() != expectedChunkIndex) {
```

File: driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java [hunk 2]
```diff
@@ -232,28 +233,28 @@ private Document getChunk(final int startChunkIndex) {
...
             throw new MongoGridFSException(format("Could not find file chunk for file_id: %s at chunk index %s.",
                     fileId, expectedChunkIndex));
         }
 
-        if (!(chunk.get("data") instanceof Binary)) {
+        if (!(chunk.get("data") instanceof BsonBinary)) {
             throw new MongoGridFSException("Unexpected data format for the chunk");
         }
-        byte[] data = chunk.get("data", Binary.class).getData();
+        byte[] data = chunk.get("data").asBinary().getData();
```
Comment (jyemin) [inline driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java:257]:
```suggestion
        byte[] data = chunk.getBinary("data").getData();
```

**Review by jyemin — CHANGES_REQUESTED** [review]
Code style suggestions.  Otherwise LGTM assuming performance is improved.

File: driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java [hunk 3]
```diff
@@ -213,17 +214,17 @@ private void discardCursor() {
...
         }
-        Document chunk = null;
+        BsonDocument chunk = null;
         if (cursor.hasNext()) {
             chunk = cursor.next();
             if (batchSize == 1) {
                 discardCursor();
             }
-            if (chunk.getInteger("n") != startChunkIndex) {
+            if (chunk.get("n").asInt32().getValue() != startChunkIndex) {
```
Comment (rozza) [inline driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java:227]:
```suggestion
            if (chunk.getInt32("n").getValue() != startChunkIndex) {
```

**Review by jyemin — APPROVED** [review]

**Review by rozza — APPROVED** [review]

**Comment by rozza** [comment]
LGTM - great find and fix, quite a subtle change / fix.

**Pull request merged**